packages/kbn-alerts-ui-shared/src/common/hooks/use_health_check.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useMemo } from 'react';
+import type { HttpStart } from '@kbn/core-http-browser';
+import { useLoadAlertingFrameworkHealth } from './use_load_alerting_framework_health';
+import { useLoadUiHealth } from './use_load_ui_health';
+import { healthCheckErrors, HealthCheckErrors } from '../apis';
+
+export interface UseHealthCheckProps {
+  http: HttpStart;
+}
+
+export interface UseHealthCheckResult {
+  isLoading: boolean;
+  healthCheckError: HealthCheckErrors | null;
+}
+
+export interface HealthStatus {
+  isRulesAvailable: boolean;
+  isSufficientlySecure: boolean;
+  hasPermanentEncryptionKey: boolean;
+}
+
+export const useHealthCheck = (props: UseHealthCheckProps) => {
+  const { http } = props;
+
+  const {
+    data: uiHealth,
+    isLoading: isLoadingUiHealth,
+    isInitialLoading: isInitialLoadingUiHealth,
+  } = useLoadUiHealth({ http });
+
+  const {
+    data: alertingFrameworkHealth,
+    isLoading: isLoadingAlertingFrameworkHealth,
+    isInitialLoading: isInitialLoadingAlertingFrameworkHealth,
+  } = useLoadAlertingFrameworkHealth({ http });
+
+  const isLoading = useMemo(() => {
+    return isLoadingUiHealth || isLoadingAlertingFrameworkHealth;
+  }, [isLoadingUiHealth, isLoadingAlertingFrameworkHealth]);
+
+  const isInitialLoading = useMemo(() => {
+    return isInitialLoadingUiHealth || isInitialLoadingAlertingFrameworkHealth;
+  }, [isInitialLoadingUiHealth, isInitialLoadingAlertingFrameworkHealth]);
+
+  const alertingHealth: HealthStatus | null = useMemo(() => {
+    if (isLoading || isInitialLoading || !uiHealth) {
+      return null;
+    }
+    if (!uiHealth.isRulesAvailable) {
+      return {
+        ...uiHealth,
+        isSufficientlySecure: false,
+        hasPermanentEncryptionKey: false,
+      };
+    }
+    if (alertingFrameworkHealth) {
+      return {
+        ...uiHealth,
+        isSufficientlySecure: alertingFrameworkHealth.isSufficientlySecure,
+        hasPermanentEncryptionKey: alertingFrameworkHealth.hasPermanentEncryptionKey,
+      };
+    }
+    return {
+      ...uiHealth,
+      isSufficientlySecure: true,
```

Comment:
It's existing behaviour but if I had to guess it's because if for some reason that endpoint doesn't return anything, we shouldn't stop the user from creating rules. 

But I removed the `true` return since im checking `alertingFrameworkHealth` now